### DISCUSSION TITLE
How to #include MayaUSD (0.6.0 install) C++

### DISCUSSION THREAD

**Discussion by MrLockness** [Autodesk/maya-usd#1164] [Q&A] [answered]
Hi guys! I'm quite fresh when it comes to C++, so forgive me if this is super obvious.

We have upgraded from Maya 2018 to Maya 2020. For Maya 2018 I compiled the code myself, and set up the environment variables that was needed. Now I need to update the tools (created using QT) to Maya 2020 and used the 0.6.0 release install on a clean windows (no previous USD versions).

My question is, how and where do I import the classes to my code?

Previously This was in my QT config file:

`LIBS		+= -LC:\PixarUSD\lib -lArch -lGf -lKind -lSdf -lTf -lUsd -lUsdGeom -lVt`
`INCLUDEPATH	+= C:\PixarUSD\include C:\PixarUSD\include\boost-1_70 C:\Python27\include `

And in my .cpp file:
`#include <pxr/usd/kind/registry.h>`
`#include <pxr/usd/usd/modelAPI.h>`
`#include <pxr/usd/usd/primRange.h>`
`#include <pxr/usd/usdGeom/xform.h>`
`#include <pxr/usd/usdGeom/metrics.h>`

However, now that I don't have a `C:/PixarUSD `folder but only the `C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd` I for the life of me do not understand how to get these functions to work again! When I look in the files located in the `C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\MayaUSD\include\` they reference to #include `<pxr/....>` too. 

The error I get when I try to compile is:
`Cannot open include file: 'pxr/usd/kind/registry.h': No such file or directory`

Again, sorry if this is super obvious  but I don't understand how to get the installer to work with c++.

Thank you so much!

**Comment by seando-adsk** [comment, marked as answer]
You will need to extract the USD devkit.zip and adjust the paths on where to find USD now. Previously you had compiled it yourself and put it in _C:\PixarUSD_. Now you want to find it in the MayaUsd plugin installation folder. If you look into the folder: 
```C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\USD``` 
you should see a file **devkit.zip** which you need to extract. This file contains the part of USD necessary to build with.

From the paths you provided above I think you would change:

```-LC:\PixarUSD\lib -> -LC:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\USD\lib```
```C:\PixarUSD\include ->C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\USD\include```
```C:\PixarUSD\include\boost-1_70 -> C:\Program Files\Autodesk\MayaUSD\Maya2020\0.6.0_202012102328-efea2cd\mayausd\USD\include\boost-1_70```

**Reply by MrLockness** [reply]
Thank you so much for your reply!
